src/main/java/org/jabref/logic/pdf/search/indexing/IndexingTaskManager.java

```diff
@@ -81,7 +99,19 @@ public void addToIndex(PdfIndexer indexer, BibEntry entry, BibDatabaseContext da
             @Override
             protected Void call() throws Exception {
                 this.updateProgress(-1, 1);
-                indexer.addToIndex(entry, databaseContext);
+                final int numFiles = entry.getFiles().size();
```

Comment:
I currently don't have the time to dive deeper into this, but one question that popup up while going over the code was: why do you need this addToIndex for an entry to be it's own background task? Each entry only has a couple of files so getting the paths of all files should be very quick, and you are not doing more in this task here if I'm not mistaken (the actual indexing is yet another task). It just adds quite a bit of overhead to have this as a separate task in my opinion.

Reply:
> the actual indexing is yet another task

No. The indexer here is the actual indexer, not the IndexingTaskManager. So the call below:
`indexer.addToIndex(entry, file, databaseContext)`
actually adds the file to the index. The
`public void addToIndex(PdfIndexer indexer, BibEntry entry, BibDatabaseContext databaseContext)` creates one task that adds all files of the given entry to the index and updates the progress in between files.